From 2b4860997a929995143262039e220c1758f559e2 Mon Sep 17 00:00:00 2001
From: James Knight <james.d.knight@live.com>
Date: Sat, 19 Oct 2019 02:45:34 -0400
Subject: [PATCH] build: Include CheckLibraryExists module (Linux)

When building under Linux, the sample's CMake definitions will use
`CHECK_LIBRARY_EXISTS` to help determine whether or not librt should be
included in the build [1]; however, the build environment may not have
included the `CheckLibraryExists` module at this time, which can result
in the following compilation error:

    CMake Error at CMakeLists.txt:<line> (CHECK_LIBRARY_EXISTS):
      Unknown CMake command "CHECK_LIBRARY_EXISTS".

Correcting this scenario by explicitly including the
`CheckLibraryExists` module in each CMakeLists.txt file which uses the
`CHECK_LIBRARY_EXISTS` function.

[1]: 881a2f3d56a9f51ec3decebf20a8534b4de662a7

Signed-off-by: James Knight <james.d.knight@live.com>
---
 CMakeLists.txt                          | 1 +
 Sample-Programs/Hologram/CMakeLists.txt | 1 +
 2 files changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 452776e42..1482aca91 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -298,6 +298,7 @@ if(WIN32)
 else()
     set(SPIRV_TOOLS_LIBRARIES ${SPIRV_TOOLS_OPT_LIB} ${SPIRV_TOOLS_LIB})
 
+    INCLUDE(CheckLibraryExists)
     CHECK_LIBRARY_EXISTS("rt" clock_gettime "" NEED_RT)
     if (NEED_RT)
         set(SPIRV_TOOLS_LIBRARIES ${SPIRV_TOOLS_LIBRARIES} rt)
diff --git a/Sample-Programs/Hologram/CMakeLists.txt b/Sample-Programs/Hologram/CMakeLists.txt
index 3051c1a1c..3348f94fd 100644
--- a/Sample-Programs/Hologram/CMakeLists.txt
+++ b/Sample-Programs/Hologram/CMakeLists.txt
@@ -97,6 +97,7 @@ add_executable(Hologram ${sources})
 target_compile_definitions(Hologram ${definitions})
 target_include_directories(Hologram ${includes})
 if("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux")
+    INCLUDE(CheckLibraryExists)
     CHECK_LIBRARY_EXISTS("rt" clock_gettime "" NEED_RT)
     if (NEED_RT)
         set(libraries ${libraries} rt)
-- 
2.23.0.windows.1

